lld/MachO/BPSectionOrderer.h

```diff
@@ -123,13 +124,17 @@ class BPSectionMacho : public BPSectionBase {
   static bool classof(const BPSectionBase *s) { return true; }
 
 private:
-  static uint64_t getRelocHash(const Reloc &reloc,
-                               const BPSectionMacho *section) {
+  static uint64_t getRelocHash(
+      const Reloc &reloc,
+      const llvm::DenseMap<const BPSectionBase *, uint64_t> &sectionToIdx) {
     auto *isec = reloc.getReferentInputSection();
     std::optional<uint64_t> sectionIdx;
-    if (isec && isec == section->getSection())
-      sectionIdx = section->getSectionIdx();
-
+    for (const auto &entry : sectionToIdx)
+      if (const auto *bpSection = llvm::dyn_cast<BPSectionMacho>(entry.first))
+        if (bpSection->getSection() == isec) {
+          sectionIdx = entry.second;
+          break;
+        }
```

Comment:
This seems to do what we want, but it is very inefficient because we are iterating over the whole map. We should instead use `DenseMap<const InputSection *, uint64_t>` so we can simply use `sectionToIdx.find(isec)`.

Reply:
I used sectionToIdx.find(isec) to handle this, but since sectionToIdx is a DenseMap<const BPSectionBase *, uint64_t> in BPSectionOrdererBase.cpp, I updated it accordingly. Is there a better way to write this?

Reply:
Yeah the key type needs to be 'InputSection'